### PR TITLE
fix: updating condition to not show empty session in user section

### PR DESCRIPTION
**Pull request by maze-runnar**
…hout email

Fixes https://github.com/fossasia/open-event-frontend/issues/5342

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/speakers.py [hunk 1]
```diff
@@ -190,7 +190,7 @@ def before_update_object(self, speaker, data, view_kwargs):
                 'Organizer access required to override email',
             )
         if (
-            data.get('is_email_overridden')
+            not data.get('is_email_overridden')
```
Comment (iamareebjamal) [inline app/api/speakers.py:193]: When did this happen?

Comment (maze-runnar) [reply app/api/speakers.py:193]: if email is overriden then there will be no email so email is null m but here are condition that is saying that if email is overriden and email is null then set it to user mail, while it should be, while email is not overriden and email is null set it to user mail

**Review by iamareebjamal — APPROVED** [review]

File: app/api/speakers.py (hunk 1 above)
Comment (iamareebjamal) [reply app/api/speakers.py:193]: @maze-runnar You changed this but didn't make email optional in schema

This is causing https://github.com/fossasia/open-event-frontend/issues/5841

Comment (maze-runnar) [reply app/api/speakers.py:193]: > @maze-runnar You changed this but didn't make email optional in schema
> 
> This is causing [fossasia/open-event-frontend#5841](https://github.com/fossasia/open-event-frontend/issues/5841)

@iamareebjamal that is related to PR https://github.com/fossasia/open-event-server/pull/7469, and I have made email optional but still showing that error.

**Pull request merged**